pkgs/unified_analytics/test/unified_analytics_test.dart

```diff
@@ -142,16 +142,38 @@ void main() {
     expect(analytics.telemetryEnabled, true,
         reason: 'Telemetry should be enabled by default '
             'when initialized for the first time');
-
     // Use the API to disable analytics
+    expect(logFile.readAsLinesSync().length, 0);
     await analytics.setTelemetry(false);
     expect(analytics.telemetryEnabled, false,
         reason: 'Analytics telemetry should be disabled');
+    expect(logFile.readAsLinesSync().length, 1,
+        reason: 'One event should have been logged for disabling analytics');
+
+    // Extract the last log item to check for the keys
+    Map<String, Object?> lastLogItem =
+        jsonDecode(logFile.readAsLinesSync().last);
+    expect((lastLogItem['events'] as List).last['name'],
+        'analytics_collection_enabled',
+        reason: 'Check on event name');
+    expect((lastLogItem['events'] as List).last['params']['status'], false,
+        reason: 'Status should be false');
 
     // Toggle it back to being enabled
     await analytics.setTelemetry(true);
     expect(analytics.telemetryEnabled, true,
         reason: 'Analytics telemetry should be enabled');
+    expect(logFile.readAsLinesSync().length, 2,
+        reason: 'Second event should have been logged toggling '
+            'analytics back on');
+
+    // Extract the last log item to check for the keys
+    lastLogItem = jsonDecode(logFile.readAsLinesSync().last);
+    expect((lastLogItem['events'] as List).last['name'],
```

Comment:
Consider extracting `(lastLogItem['events'] as List).last` as a helper or extension method, given that it occurs 4 times in this method alone.